src/nbl/video/CVulkanCommandPool.cpp

```diff
@@ -12,6 +12,18 @@ CVulkanCommandPool::~CVulkanCommandPool()
     vk->vk.vkDestroyCommandPool(vulkanDevice->getInternalObject(), m_vkCommandPool, nullptr);
 }
 
+void CVulkanCommandPool::reset()
+{
+	// Free everything in the memory pool from all command buffers
+	// TODO figure out if this is the best way to do it
```

Comment:
not only do we need to free the mempool, we need to run the destructors of all objects on the tape :(

Related to https://github.com/Devsh-Graphics-Programming/Nabla/pull/408#discussion_r940693463

Reply:
also `mempool`, etc. should move to the base probably